### PR TITLE
Fixed Barcode Scanner

### LINKED ISSUE #2353: Collect crashes when click on "Get Barcode" or "Scan code from other device"
#### Software and hardware versions 
Collect v1.15..x, Android all device used...

#### Problem description
Collect is crashing when using Barcode widget and when try to import/export settings by the barcode.

#### Steps to reproduce the problem
1. Go to Import/export settings and click on "Scan code from other device"
2. Use All widgets form and Barcode widget.

#### Expected behavior
The user should be able to use the barcode widget and import/export settings.

#### Other information 
It is a regression and it's probably caused by changes from: https://github.com/opendatakit/collect/pull/2309

@opendatakit-bot label "regression"
@opendatakit-bot label "high priority"

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #2353 

#### What has been done to verify that this works as intended?
I tested scanning barcodes using BarcodeWidget and scanning QR code with settings.

#### Why is this the best possible solution? Were any other approaches considered?
We need ids I added back.

#### Are there any risks to merging this code? If so, what are they?
No.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [ ] run `./gradlew pmd checkstyle lint findbugs` and confirmed all checks still pass.
- [ ] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: config/lint.xml [hunk 1]
```diff
@@ -6,6 +6,9 @@
     <issue id="AppCompatResource" severity="warning"/>
     <issue id="StaticFieldLeak" severity="warning"/>
     <issue id="UnusedIds" severity="error"/>
+    <issue id="UnusedIds" severity="error">
+        <ignore path="res/layout/custom_barcode_scanner.xml"/>
```
Comment (lognaturel) [inline config/lint.xml]: Good catch! We should try to make ignores as local as they can possibly be so we can still benefit from the check in the rest of the file. In this case, you should be able to request it directly in the xml - https://developer.android.com/studio/write/lint#src. If that syntax doesn't work, there's also a comment syntax - https://stackoverflow.com/a/23964496. Then your comment can say something like "ID used by zxing library".

Comment (grzesiek2010) [reply config/lint.xml]: Yeah, you are right, thanks. FIxed.

**Review by vishnus1224 — APPROVED** [review]

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Thank you for helping with review, @vishnus1224! Much appreciated! 🎉

**Comment by mmarciniak90** [comment]
**Tested with success!**

Verified on Android: 4.1, 4.2, 4.4, 5.1, 6.0, 7.0, 8.1
Verified cases:
* Scan QR code settings
* Scan QR code in Barcode widget

@opendatakit-bot label "behavior verified"
@opendatakit-bot unlabel "needs testing"

**Pull request merged**